File: src/BaseHmiTypes/Screens/Base/HmiSymbolContainer.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiSymbolContainer : HmiCustomWidgetContainer
{
    public HmiSymbolContainer()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiSymbolContainer;
    }

    public HmiProperty<HmiSymbolFillColorMode>? FillColorMode { get; set; }

    public HmiProperty<HmiSymbolFlipMode>? Flip { get; set; }

    public HmiProperty<bool>? FixedAspectRatio { get; set; }

    public HmiProperty<HmiImageSource>? Image { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiSymbolContainer.ts
import { HmiCustomWidgetContainer } from "./HmiCustomWidgetContainer.js";
import { HmiImageSource } from "./HmiImageSource.js";
import { HmiProperty } from "./HmiProperty.js";
import { HmiSymbolFillColorMode } from "./HmiSymbolFillColorMode.js";
import { HmiSymbolFlipMode } from "./HmiSymbolFlipMode.js";
import { HmiObjectType } from "./HmiObjectType.js";

export class HmiSymbolContainer extends HmiCustomWidgetContainer {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiSymbolContainer;
  }

  fillColorMode?: HmiProperty<HmiSymbolFillColorMode>;
  flip?: HmiProperty<HmiSymbolFlipMode>;
  fixedAspectRatio?: HmiProperty<boolean>;
  image?: HmiProperty<HmiImageSource>;
}
